### PR TITLE
Add tests of tex{Sub}Image3D taking sub-sources of images.

### PR DESCRIPTION
**Pull request by kenrussell**
Remove tests of obsolete entry points and redundant tests.

### CODE REVIEW

**Comment by kenrussell** [comment]
@zhenyao @kainino0x @jdashg please review.

Tested with Chromium's implementation in https://codereview.chromium.org/2482793005/ .

The new resource is a 4x4 PNG (also the Photoshop original) containing 2x2 squares of red, green, blue, and cyan. It's used to select 2x4 sub-rectangles and upload them to 2x2 3D and 2D_ARRAY textures.

File: sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image.js [hunk 1]
```diff
@@ -72,93 +91,167 @@ function generateTest(internalFormat, pixelFormat, pixelType, prologue, resource
...
+    function runRedGreenBlueCyanTest(image) {
+        function runOneIteration(image, useTexSubImage3D, flipY, bindingTarget,
+                                 depth, sourceSubRectangle, unpackImageHeight,
+                                 tTextureCoord, topColor, bottomColor, program)
+        {
+            uploadImageToTexture(image, useTexSubImage3D, flipY, bindingTarget,
+                                 depth, sourceSubRectangle, unpackImageHeight);
+            var tCoordLocation = gl.getUniformLocation(program, 'uTCoord');
+            if (tCoordLocation < 0) {
+                testFailed('Shader incorrectly set up; couldn\'t find uTCoord uniform');
```
Comment (zhenyao) [inline sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image.js]: nit: return in this case

File: sdk/tests/js/tests/tex-image-and-sub-image-utils.js [hunk 2]
```diff
@@ -196,17 +198,18 @@ var TexImageUtils = (function() {
...
   var simple3DIntTextureFragmentShaderES3 = [
     '#version 300 es',
     'precision mediump float;',
     'uniform mediump isampler3D tex;',
     'in vec2 texCoord;',
+    'uniform float uTCoord;',
     'out vec4 fragData;',
     'void main() {',
-    '    ivec4 data = texture(tex, vec3(texCoord, 0.0));',
+    '    ivec4 data = texture(tex, vec3(texCoord, uTCoord));',
```
Comment (zhenyao) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: The third component should either be r or p (renamed not to confuse with color component r).  Name it t is really confusing.

File: sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image.js (hunk 1 above)
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-3d-with-image.js]: Fixed.

**Comment by kenrussell** [comment]
This patch passed the Travis CI checks earlier so I'm merging this now; it looks like it could be a while before the Travis queue is available.

**Pull request merged**